src/cch/orders_db.rs

```diff
@@ -1,18 +1,71 @@
 use std::collections::HashMap;
 
-use super::{error::CchDbError, SendBTCOrder};
+use super::{error::CchDbError, ReceiveBTCOrder, SendBTCOrder};
 
 // TODO: persist orders
 #[derive(Default)]
 pub struct CchOrdersDb {
     /// SendBTCOrder map by payment hash
     send_btc_orders: HashMap<String, SendBTCOrder>,
+    receive_btc_orders: HashMap<String, ReceiveBTCOrder>,
 }
 
 impl CchOrdersDb {
     pub async fn insert_send_btc_order(&mut self, order: SendBTCOrder) -> Result<(), CchDbError> {
```

Comment:
Since the key is just the payment hash here, I am wondering if an adversary can create a fake `AddTlc` order to nullify a valid send_btc request.

Reply:
Overwriting is not allowed. Real user can retry a new payment hash.